cli/config/credentials/env_store.go

```diff
@@ -0,0 +1,79 @@
+package credentials
+
+import (
+	"encoding/json"
+	"maps"
+	"os"
+
+	"github.com/docker/cli/cli/config/types"
+)
+
+type envStore struct {
+	envCredentials map[string]types.AuthConfig
+	fileStore      Store
+}
+
+func (e *envStore) Erase(serverAddress string) error {
+	// We cannot erase any credentials in the environment
+	// let's fallback to the file store
+	if err := e.fileStore.Erase(serverAddress); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (e *envStore) Get(serverAddress string) (types.AuthConfig, error) {
+	authConfig, ok := e.envCredentials[serverAddress]
+	if !ok {
+		return e.fileStore.Get(serverAddress)
+	}
+	return authConfig, nil
+}
+
+func (e *envStore) GetAll() (map[string]types.AuthConfig, error) {
+	credentials := make(map[string]types.AuthConfig)
+	fileCredentials, err := e.fileStore.GetAll()
+	if err == nil {
+		credentials = fileCredentials
+	}
+
+	// override the file credentials with the env credentials
+	maps.Copy(credentials, e.envCredentials)
+	return credentials, nil
+}
+
+func (e *envStore) Store(authConfig types.AuthConfig) error {
+	// We cannot store any credentials in the environment
+	// let's fallback to the file store
+	return e.fileStore.Store(authConfig)
+}
```

Comment:
> Login would store credentials in memory

Hm.. or indeed use the fallback; I guess either _could_ make sense here 🤔 (tough choices!)